railties/lib/rails/commands/server/server_command.rb

```diff
@@ -89,18 +41,6 @@ def middleware
       Hash.new([])
     end
 
-    def default_options
```

Comment:
The reason I didn't pursue writing these options using Thor in the first place was that `Rails::Server` interops with `Rack::Server` in certain ways as seen by the `super` here.

Thus, since some of the methods you removed were also public, I figured they were part of `Rails::Server`'s public API and to remove them we'd need to go through a deprecation cycle.

cc @rafaelfranca